### PR TITLE
Refactor efc images

### LINKED ISSUE #3199: [FEATURES]: refactor efcruntime images and support cpfs
**What feature you'd like to add:**
refactor efcruntime images and support cpfs

**Why is this feature needed:**

### PR DESCRIPTION
**Pull request by yangjun289519474**
### Ⅰ. Describe what this PR does
refactor efcruntime images and support cpfs

### Ⅱ. Does this pull request fix one issue?

fixes #3199 

### Ⅴ. Special notes for reviews

### CODE REVIEW

**Comment by cheyang** [comment]
@yangjun289519474 please fix the conflict. thanks.

File: pkg/ddc/efc/utils.go [hunk 1]
```diff
@@ -172,34 +173,50 @@
...
 		mount.MountPoint = mount.MountPoint + "/"
 	}
 
-	if !strings.HasPrefix(mount.MountPoint, MountPointPrefix) {
-		return info, fmt.Errorf("invalid mountpoint prefix for EFCRuntime name:%s, namespace:%s, mountpoint:%s", e.name, e.namespace, mount.MountPoint)
-	} else {
-		info.MountPoint = strings.TrimPrefix(mount.MountPoint, MountPointPrefix)
-	}
+	if strings.HasPrefix(mount.MountPoint, NasMountPointPrefix) {
+		reg, err := regexp.Compile(`^(nfs://)([a-z0-9-]+).([a-z0-9-]+).nas.aliyuncs.com:`)
```
[the remark this answers is not in the document]
Comment (yangjun289519474) [reply pkg/ddc/efc/utils.go]: fix

File: pkg/ddc/efc/utils.go [hunk 2]
```diff
@@ -172,34 +173,50 @@
...
-	} else {
-		info.FileSystemId = strings.Split(info.MountPoint, "-")[0]
-	}
+		info.MountPoint = strings.TrimPrefix(mount.MountPoint, NasMountPointPrefix)
+		info.MountPointPrefix = result[0][1]
+		info.FileSystemId = strings.Split(result[0][2], "-")[0]
+		info.ServiceAddr = result[0][3]
+		info.DirPath = strings.TrimPrefix(mount.MountPoint, result[0][0])
+	} else if strings.HasPrefix(mount.MountPoint, CpfsMountPointPrefix) {
+		reg, err := regexp.Compile(`^(cpfs://)([a-z0-9-]+).([a-z0-9-]+).cpfs.aliyuncs.com:/share`)
```
[the remark this answers is not in the document]
Comment (yangjun289519474) [reply pkg/ddc/efc/utils.go]: fix

**Comment by yangjun289519474** [comment]
> @yangjun289519474 please fix the conflict. thanks.

done

**Comment by yangjun289519474** [comment]
> @yangjun289519474 please fix the conflict. thanks.

done!

**Review by cheyang — APPROVED** [review]
/lgtm
/approve

**Pull request merged**